File: examples/man-test/art/directions.js
const cartoonLib = package('@funcdraw/testlib')?.cartoon ?? {};
const stickmanModule = cartoonLib?.stickman;
const createStickman =
  typeof stickmanModule?.static === 'function'
    ? stickmanModule.static
    : typeof stickmanModule === 'function'
      ? stickmanModule
      : () => ({ graphics: [] });

const defaultPose = typeof createStickman.skeleton === 'function' ? createStickman.skeleton() : null;
const defaultY = Array.isArray(defaultPose?.position) ? defaultPose.position[1] : 10.5;
const groundY = 0;
const lineup = [
  { x: -48, direction: 'left' },
  { x: -16, direction: 'front' },
  { x: 16, direction: 'back' },
  { x: 48, direction: 'right' }
];

const heroes = lineup.map(({ x, direction }) =>
  createStickman({
    position: [x, groundY + 22],
    measurements: {
      torso: { direction, height: 22, width: 12 },
      head: { direction, verticalExtent: 9 },
      legs: {
        left: { effectorCoordinate: [-4, -22] },
        right: { effectorCoordinate: [4, -22] }
      },
      hands: {
        left: { effectorCoordinate: [-7.8, 4.7] },
        right: { effectorCoordinate: [7.8, 4.7] }
      }
    }
  })
);

const labels = lineup.map(({ x, direction }) => ({
  type: 'text',
  text: direction,
  position: [x, -6],
  fill: '#0f172a',
  fontSize: 12,
  align: 'center'
}));

const baseline = {
  type: 'line',
  from: [-56, groundY],
  to: [56, groundY],
  stroke: '#94a3b8',
  width: 0.5
};

return {
  view: { left: -40, bottom: -30, right: 70, top: 90 },
  graphics: [
    baseline,
    ...heroes.flatMap((hero) => (hero && Array.isArray(hero.graphics) ? hero.graphics : [])),
    ...labels
  ]
};

File: examples/man-test/art/directions.fs
{
  stickman:package("@funcdraw/testlib").cartoon.stickman;
  createStickman:stickman.static;

  lineupX:constants.directions.lineup;
  lineupDirections:["left", "front", "back", "right"];
  groundY:0;

  defaultPose:createStickman({});
  anchorY:defaultPose.skeleton.position[1];

  createHero:(x, direction)=> createStickman({
    position:[x, anchorY];
    measurements:{
      torso:{ direction:direction; height:22; width:12 };
      head:{ direction:direction; verticalExtent:9 };
      legs:{
        left:{ effectorCoordinate:[-4, -22] };
        right:{ effectorCoordinate:[4, -22] };
      };
      hands:{
        left:{ effectorCoordinate:[-7.8, 4.7] };
        right:{ effectorCoordinate:[7.8, 4.7] };
      };
    };
  });

  heroes:[
    createHero(lineupX[0], lineupDirections[0]),
    createHero(lineupX[1], lineupDirections[1]),
    createHero(lineupX[2], lineupDirections[2]),
    createHero(lineupX[3], lineupDirections[3])
  ];

  labels:[
    { type:"text"; text:lineupDirections[0]; position:[lineupX[0], constants.directions.labelY]; fill:"#0f172a"; fontSize:12; align:"center" },
    { type:"text"; text:lineupDirections[1]; position:[lineupX[1], constants.directions.labelY]; fill:"#0f172a"; fontSize:12; align:"center" },
    { type:"text"; text:lineupDirections[2]; position:[lineupX[2], constants.directions.labelY]; fill:"#0f172a"; fontSize:12; align:"center" },
    { type:"text"; text:lineupDirections[3]; position:[lineupX[3], constants.directions.labelY]; fill:"#0f172a"; fontSize:12; align:"center" }
  ];

  baseline:{
    type:"line";
    from:[-constants.directions.baselineHalfSpan, groundY];
    to:[constants.directions.baselineHalfSpan, groundY];
    stroke:"#94a3b8";
    width:0.5;
  };

  heroGraphics:heroes[0].graphics + heroes[1].graphics + heroes[2].graphics + heroes[3].graphics;

  eval {
    view:{ left:-40; bottom:-30; right:70; top:90 };
    graphics:[baseline, heroGraphics, labels];
  };
}
